### DISCUSSION TITLE
Analog pots go crazy at start and then suddenly everything works perfect

### DISCUSSION THREAD

**Discussion by eurorackworld** [shanteacontrols/OpenDeck#236] [Q&A] [answered]
I have a really strange issue that 'solves itself' if you keep the controller connected for a short period. I'll explain the issue below.

**Overview of the controller**
<img width="443" alt="image" src="https://github.com/shanteacontrols/OpenDeck/assets/29734704/a3f1e4ec-2e11-4a0c-bf67-d6db8fa28d0d">
<img width="400" alt="image" src="https://github.com/shanteacontrols/OpenDeck/assets/29734704/a4a627d9-1955-4824-af89-93a769617f00">
It is a pcb with 36 potentiometers (regular RV09, 3pins), all connected to 3x a MUX chip (HC4067). I also have via 2x 220ohm resistors midi connection on the side (5V - ground - TX), in this example this is disconnected (see 3 wires). Power is all 3V3 standard from the RPI Pico, only midi is connected to the 5V pin or else it won't work.

On the RPI Pico there is custom firmware, which I also have been using on PCB's before which were more simple and never had any issues. This is an evolution of those PCB's , mainly the MIDI connection with the resistors is new. 

**Issue**
I built this controller and after the first test all worked fine. Then later another test suddenly shows the analog controls going crazy, values all over the place from all potentiometers. Fully disconnected this is still the case. Today I noticed that for 2 controllers I made with this board (both having the same issue) that after a while (sometimes max 10s), the values start to get less crazy and suddenly stable again. Then the controller works fine without any issues. 

When unplugging and plugging in again the Micro USB the whole issue repeats itself, having crazy going pots with values, slowly 'dying out' and then quiet and everything works as normal. 

Also noticed that during the test for example the buttons connected to it and the midi out worked fine. Plus sometimes when touching (on random places on the pcb) the values start 'dying out' sooner and the controller sometimes instantly works normal again. 

**Question**
So I can share config, but the controller does work fine so it cannot (at least shouldn't) be the firmware. I used the same for the older PCB I made and those had never these kind of issues. But I'm wondering if anyone recognize this electronical behaviour, as I don't think it has anything to do with the Opendeck software, but just something on how I made the setup on this board. 

Any thoughts are welcome, as I cannot ask my customers to first wait or start internally touching their controller before they can use it :)

p.s. I've read the note on 3V3 challenges for the 4067 but I never had this issue before. Also the previous board had the same MUX chips on it. I'll try to measure the voltages that go through the PCB as maybe it just drop below a certain point making the mux go haywire, and if the voltage stabilizes is gets ok again. Just assumptions as I am not a real in depth electronics expert.

**Comment by paradajz** [comment, marked as answer]
> only midi is connected to the 5V pin or else it won't work.

Pico is a 3.3V device - if you connect 5V to it's inputs you're pretty much guaranteed to damage the MCU. So I'm not sure what do you mean with this?

This is, for example, the schematic I'm using for MIDI on the OpenDeck M/L boards which use RP2040 MCU:

![image](https://github.com/shanteacontrols/OpenDeck/assets/2544094/189e797c-a30c-4603-8a2d-b62953153e54)

Other than that, off the top of my head - are you using ground plane on your PCB? Lack of it can cause issues such as you've described.

**Reply by eurorackworld** [reply]
Big thanks! That ground plane thing I don't have, sounds like the issue indeed. Will check!

p.s. Did you get an email from me, not sure if it arrived?

**Reply by eurorackworld** [reply]
Oh and btw, I did use the Vbus connection on the PICO with 5V to the midi connector as that required 5V to work, and used 220ohm resistors on both the TX as the 5V wire. Not sure on the pins you're mentioning here, as I use the middle 3 pins. I believe 2 is 220ohm to 5V, 3 is the GND and 4 is 220ohm to TX. Works great so not sure if yours is an alternative option, or mine is just a bad one :P

**Reply by eurorackworld** [reply]
Received new PCB's with a ground plane on both sides, so far testing shows all issues are gone (although the test setup is with a limited set of controls, nevertheless it looks better than before).

**Comment by paradajz** [comment]
> p.s. Did you get an email from me, not sure if it arrived?

I'll check.

> Oh and btw, I did use the Vbus connection on the PICO with 5V to the midi connector as that required 5V to work, and used 220ohm resistors on both the TX as the 5V wire. Not sure on the pins you're mentioning here, as I use the middle 3 pins. I believe 2 is 220ohm to 5V, 3 is the GND and 4 is 220ohm to TX. Works great so not sure if yours is an alternative option, or mine is just a bad one :P

Can you share the schematics you've used for the MIDI part?

**Reply by eurorackworld** [reply]
I don't have any .... 

But I mixed up the numbers as I was matching your image to my DIN socket, but your numbers are matching my numbers only the midi  connector numbering on the socket is not 1,2,3,4,5 but 3,5,2,4,1, , and therefore you also use the middle 3 pins of the socket.. Sorry for the confusion. Only difference is the 220 ohm on both pins for me, instead of 33 and 10.

So I have:
RPI Pico PIN40/Vbus -> 220ohm resistor -> midi din pin 4  
RPI Pico GNd -> midi din pin 2
RPI Pico PIN1/GPO -. 220ohm resistor -> midi din pin 5

as shown here: https://docs.arduino.cc/built-in-examples/communication/Midi 
with these schematics:
![image](https://github.com/shanteacontrols/OpenDeck/assets/29734704/55800117-9680-48c3-9e4b-f38037c9e74d)
![image](https://github.com/shanteacontrols/OpenDeck/assets/29734704/6e11b78f-2e82-45e2-b0cc-94d61955e14d)

but I also see variants using diodes as well, so guess there are many ways to rome :P

**Reply by paradajz** [reply]
Again, you are ignoring the fact that the Pico is 3V device and you're using schematic made for 5V signaling.
